File: Python/Zadanie 2/szachy_python/pieces/king.py
from piece import Piece

class King(Piece):
    def symbol(self):
        return '♔' if self.color == 'white' else '♚'

    def is_valid_move(self, board, start, end):
        dx = abs(end[0] - start[0])
        dy = abs(end[1] - start[1])
        if max(dx, dy) > 1:
            return False
        target = board.get_piece(end)
        return target is None or target.color != self.color

File: C#/Zadanie 2/Szachy/King.cs
namespace Szachy;

public class King : Piece
{
    public King(PieceColor color) : base(color) {}

    public override char Symbol => Color == PieceColor.White ? 'K' : 'k';

    public override bool IsValidMove((int x, int y) from, (int x, int y) to, Board board)
    {
        int dx = Math.Abs(to.x - from.x);
        int dy = Math.Abs(to.y - from.y);

        // Król może się poruszyć o 1 pole we wszystkich kierunkach
        if (dx <= 1 && dy <= 1)
        {
            var target = board.Squares[to.x, to.y];
            if (target == null || target.Color != this.Color)
                return true;
        }
        return false;
    }
}